OlinkAnalyze/R/Olink_normalization.R

```diff
@@ -53,22 +56,35 @@
 #'
 #' #Subset normalization:
 #' # Find a suitable subset of samples from both projects, but exclude Olink controls
-#' df1_sampleIDs <- (npx_df1 %>%
-#'     dplyr::filter(!grepl("control", SampleID, ignore.case=TRUE)) %>%
+#' # and samples which do not pass QC.
+#' df1_sampleIDs <- npx_df1 %>%
+#'     dplyr::filter(QC_Warning == 'Pass') %>%
+#'     dplyr::filter(!str_detect(SampleID, 'CONTROL_SAMPLE')) %>%
 #'     dplyr::select(SampleID) %>%
-#'     dplyr::distinct())$SampleID
-#' df2_sampleIDs <- (npx_df2 %>%
-#'     dplyr::filter(!grepl("control", SampleID, ignore.case=TRUE)) %>%
+#'     unique() %>%
+#'     dplyr::pull(SampleID)
+#' df2_sampleIDs <- npx_df2 %>%
+#'     dplyr::filter(QC_Warning == 'Pass') %>%
+#'     dplyr::filter(!str_detect(SampleID, 'CONTROL_SAMPLE')) %>%
```

Comment:
```suggestion
#'     dplyr::filter(!stringr::str_detect(SampleID, 'CONTROL_SAMPLE')) %>%
```

{stringr} is not attached in the example, thus we need to give explicit namespace